pyodide_build/tests/test_venv.py

```diff
@@ -0,0 +1,220 @@
+import os
+import shutil
+import subprocess
+
+import pytest
+
+from pyodide_build.out_of_tree import venv
+from pyodide_build.xbuildenv import CrossBuildEnvManager
+
+
+@pytest.fixture(scope="session")
+def base_test_dir(tmp_path_factory):
+    """Create a session-wide directory that persists for all tests."""
+    base_dir = tmp_path_factory.getbasetemp() / "pyodide_test_base"
+    base_dir.mkdir(exist_ok=True)
+
+    venv_path = base_dir / "test_venv"
+    venv_path.mkdir(exist_ok=True)
+
+    cwd = os.getcwd()
+    os.chdir(str(base_dir))
+
+    xbuildenv_test_name = ".pyodide-xbuildenv-for-testing"
+
+    manager = CrossBuildEnvManager(xbuildenv_test_name)
+    manager.install()
+
+    os.chdir(cwd)
+
+    # Clean the venv before yielding, but preserve the xbuildenv
+    # as we can reuse it between the tests
+    if venv_path.exists():
+        for item in venv_path.iterdir():
+            if not str(item.name).startswith(xbuildenv_test_name):
+                if item.is_dir():
+                    shutil.rmtree(item)
+                else:
+                    item.unlink()
+    yield base_dir
+    shutil.rmtree(base_dir)
+
+
+@pytest.mark.parametrize(
+    "options,expected_calls",
+    [
+        ([], []),
+        (["--clear"], ["--clear"]),
+        (["--no-vcs-ignore"], ["--no-vcs-ignore"]),
+        (["--pip", "23.0.1"], ["--pip", "23.0.1"]),
+        (["--no-setuptools"], ["--no-setuptools"]),
+        (["--no-wheel"], ["--no-wheel"]),
+        (["--no-periodic-update"], ["--no-periodic-update"]),
+        # TODO: enable when they are supported
+        # (["--symlink-app-data"], ["--symlink-app-data"]),
+        # (["--copies"], ["--copies"]),
+    ],
+)
+def test_venv_cli_args(monkeypatch, options, expected_calls, tmp_path):
+    """Test that CLI options are correctly passed to virtualenv."""
+    captured_args = None
+    temp_venv_path = tmp_path / "test_venv"
+
+    def mock_session_via_cli(args):
+        nonlocal captured_args
+        captured_args = args
+
+        class MockSession:
+            def __init__(self):
+                self.creator = type("MockCreator", (), {"dest": str(temp_venv_path)})
+
+            def run(self):
+                # create the directory to avoid cleanup issues
+                temp_venv_path.mkdir(exist_ok=True)
+
+        return MockSession()
+
+    temp_venv_path.mkdir(exist_ok=True)
+
+    # Mock most of the functions called by create_pyodide_venv
+    # as we just need to check the arguments passed to it
+    monkeypatch.setattr("virtualenv.session_via_cli", mock_session_via_cli)
```

Comment:
(no need to address in this PR)

It is a bit sad that we need so many mocks there. I guess we can refactor the `venv` code better to provide a more high level interface.

Reply:
I agree!